pkg/chezmoi/sourcestate.go

```diff
@@ -651,8 +659,20 @@ func (s *SourceState) Encryption() Encryption {
 
 // ExecuteTemplateData returns the result of executing template data.
 func (s *SourceState) ExecuteTemplateData(name string, data []byte) ([]byte, error) {
+	leftDelim := "{{"
+	rightDelim := "}}"
+	extension := strings.TrimPrefix(path.Ext(strings.TrimSuffix(name, ".tmpl")), ".")
+	delims := s.delims[extension]
+	if len(delims.Left) > 0 {
+		leftDelim = delims.Left
+	}
+	if len(delims.Right) > 0 {
+		rightDelim = delims.Right
+	}
+
 	tmpl, err := template.New(name).
 		Option(s.templateOptions...).
+		Delims(leftDelim, rightDelim).
```

Comment:
You should be able to pass `delims.Left` and `delims.Right` here, as, if either is empty, it will revert to the default (`{{` or `}}`). This will eliminate the `leftDelim` and `rightDelim` local variables completely.

Reply:
I had it like that originally, but I recall it panicking when `delims` was `nil`. I'll double check and change appropriately.

Reply:
I think that if you were to make it so that it’s `map[string]Delim`, it would work (I could be wrong; I’m not really a gopher). At a minimum, though, if there *are* no delimiters specified, we should always `make(map[string]*Delim)`.